### ISSUE TITLE
Pseudobulk analysis

### ISSUE THREAD

**Issue by klgoss** [satijalab/seurat#6892]
Hello, 

I was wondering if it is possible to pseudobulb scRNAseq data. I tried both `AggregateExpression` as well as `Seurat:::PseudobulkExpression`. I'd like to generate a PCA plot as well as differential gene expression but I'm not able to get that to work. Is this possible? 

Thanks in advance for any insight!!

**Comment by yuhanH** [comment]
After running `AggregateExpression` with the setting ` return.seurat = TRUE`, you will get a seurat object. 
Then you can use any Seurat functions to process the bulk object. 
Here is an example:
```
library(SeuratData)
data("pbmc3k")
library(Seurat)
library(dplyr)
# make a fake individual information for pbmc3k
pbmc3k$donor <- sample(x = c('A', 'B','C'), size = ncol(pbmc3k), replace = TRUE)
pbmc3k <- NormalizeData(pbmc3k)

# pseudo-bulk by per donor per cell type
pbmc3k.bulk <- AverageExpression(pbmc3k, return.seurat = TRUE, group.by = c('donor','seurat_annotations'))

pbmc3k.bulk <- NormalizeData(pbmc3k.bulk)%>% FindVariableFeatures() %>% ScaleData()%>% RunPCA(npcs = 10)
pbmc3k.bulk$celltype <- sapply(strsplit(Cells(pbmc3k.bulk), split = '_'), '[', 2)

DimPlot(pbmc3k.bulk, group.by = 'celltype', label = TRUE,  reduction = 'pca')
```

**Comment by jiangzh-coder** [comment]
Thanks lot! It is very interesting, but i have 1 question:

if data is already pseudobulk object from scRNAseq data with logCPM value, how can i change it back to a seurat object with counts value? Can i still use above method to turn data back to a seurat object? My data is normalized to become a pseudobulk data as following: "Normalizing count data
After excluding poor quality cells, we normalized the sequencing depth of each cell by dividing
each cell’s counts by the total counts in that cell, resulting in a matrix where the entries represent
the proportion of a cell’s reads allocated to each gene (i.e. values in the range [0,1]). To estimate
a library size for each dataset, we summed the total counts in each cell, and then we took the
median as the library size for the dataset. Next, we multiplied the proportions by the library size
to get a count matrix that was normalized for sequencing depth. Finally, we transformed the
normalized count matrix with log2(1 + count). We referred to this log-transformed quantity in the
figures as log2CPM.

We created pseudobulk expression (L. Lun, Bach, and Marioni 2016) for the cells in a
cluster for each donor such that the pseudobulk matrix had one row for each gene and one column
for each cluster from each patient.
We normalized the pseudobulk counts to log2CPM as
described in the previous section. Then we use limma::lmFit() to test for differential gene
expression with the log2CPM pseudobulk matrix (Ritchie et al. 2015). We also use
presto::wilcoxauc() to compute the area under receiver operator curve (AUROC or AUC) for the
log2CPM value of each gene as a predictor of the cluster membership for each cluster
(Korsunsky, Nathan, et al. 2019).
"
thanks best wishes
J.
